Agents/EnergyPerformanceCertificateAgent/agent/kgutils/querytemplates.py

```diff
@@ -273,6 +273,40 @@ def get_ocgml_crs():
     return query
 
 
+def get_all_pure_inputs(property_iris: list = []):
+    # Retrieve IRIs of all (potential) pure inputs instantiated/updated by EPC agent, i.e.
+    # - OBE properties (Building, Flat, Property)
+    # - OBE floor area
+    # - OBE postal code
+
+    # Remove any potential None values and duplicates
+    property_iris = [iri for iri in property_iris if iri]
```

Comment:
This line will slice the IRI if it's accidentally passed in as a string. You probably want to add something before this line like the below:
```python
if not isinstance(property_iris, list):
    property_iris = [property_iris]
```

Reply:
Addressed by commit [9b054aa](https://github.com/cambridge-cares/TheWorldAvatar/pull/629/commits/9b054aa4aa7daea4bb9a70311c5782677c7ce8fb)